src/Plugin/Action/EntityCreate.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\EntityCreate.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a generic 'Create a new entity' action.
+ *
+ * @Action(
+ *   id = "rules_entity_create",
+ *   deriver = "Drupal\rules\Plugin\Action\EntityCreateDeriver",
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class EntityCreate extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The entity type.
+   *
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * Constructs an EntityCreate object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityManagerInterface $entity_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityManager = $entity_manager;
+    $this->entityType = $plugin_definition['entity_type'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t("Create a new $this->entityType entity");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $values = $this->getContextValues();
+    $entity = $this->entityManager->getStorage($this->entityType)->create($values);
```

Comment:
So we actually don't need the entity manager but only the entity storage? Then we should inject the entity storage instead?